### DISCUSSION TITLE
How to show stale content while fresh content is loading from server component

### DISCUSSION THREAD

**Discussion by donaldshen** [vercel/next.js#59397] [Help]
### Summary

Basically, I have a similar scenario as this example: https://react.dev/reference/react/Suspense#showing-stale-content-while-fresh-content-is-loading. It uses the [useDeferredValue](https://react.dev/reference/react/useDeferredValue) hook to achieve the result, and inside using `use()` hook to fetch the data from client side.

But in Nextjs world, we only have the "switch to fallback when streaming" example https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming#example. There's no other example in the [Suspense](https://react.dev/reference/react/Suspense) document, especially the "show stale content while loading fresh content" one.

After trying for a bit myself, I don't see there's a way to integrate the `useDeferredValue` hook with server component pattern, nor getting any answers from the internet.

**Comment by marinabgarcia** [comment]
I'm having the same problem. Do you have any updates regarding this? Thanks!

**Reply by donaldshen** [reply]
sadly, no🥶

**Comment by adamwgriffin** [comment]
I saw the same docs on `useDeferredValue` and have yet to figure out a solution either.
